objects/src/accounts/mod.rs

```diff
@@ -216,6 +216,25 @@ impl Account {
         Ok(())
     }
 
+    /// Sets the nonce of this account to the specified nonce value.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The new nonce is smaller than the actual account nonce
+    /// - The new nonce is equal to the actual account nonce
+    pub fn set_nonce(&mut self, nonce: Felt) -> Result<(), AccountError> {
```

Comment:
Let's use this function instead of lines 207 - 213 above.

Reply:
Updated.

Also added some questions here: https://github.com/0xPolygonMiden/miden-node/issues/223#issuecomment-1943707465